File: golang_solution/stack/1190.java
import java.util.*;

class Solution1190 {
    int i = 0;

    public String reverseParentheses(String s) {
        return recur(s);
    }

    public String recur(String s) {
        StringBuilder sb = new StringBuilder();
        while (i < s.length()) {
            if (s.charAt(i) == '(') {
                i++;
                sb.append(recur(s));
            } else if (s.charAt(i) == ')') {
                return reverse(sb.toString());
            } else {
                sb.append(s.charAt(i));
            }
            i++;
        }
        return sb.toString();
    }

    public static String reverse(String str) {
        char[] chars = str.toCharArray();
        String reverse = "";
        for (int i = chars.length - 1; i >= 0; i--) {
            reverse += chars[i];
        }
        return reverse;
    }
}

File: golang_solution/stack/1190.go
package stack

func reverseParentheses(s string) string {
	i := 0
	var recur func() string
	recur = func() string {
		str := ""
		for i < len(s) {
			if s[i] == '(' {
				i++
				str += recur()
			} else if s[i] == ')' {
				return reverse(str)
			} else {
				str += string(s[i])
			}
			i++
		}
		return str
	}
	return recur()
}

func reverse(a string) string {
	n := len(a)
	as := []byte(a)
	i, j := 0, n-1
	for i < j {
		as[i], as[j] = as[j], as[i]
		i++
		j--
	}
	return string(as)
}
